app/controllers/lab_tests_controller.rb

```diff
@@ -80,20 +88,55 @@ def destroy
 
   private
 
-  # Use callbacks to share common setup or constraints between actions.
+  def handle_success_response
+    flash[:notice] = t('.success')
+    redirect_to(@health_record)
+  end
+
+  def handle_error_response(message = nil)
+    flash.now[:alert] = message if message
+    load_error_dependencies
+    render :new, status: :unprocessable_entity
+  end
+
+  def determine_redirect_path
+    @health_record || @lab_test
+  end
+
+  def load_error_dependencies
+    return if @lab_test.biomarker_id.blank?
+
+    @selected_biomarker = Biomarker.find(@lab_test.biomarker_id)
```

Comment:
Why do you query biomarker table here again if you have already had all the biomarkers from the         @biomarkers = Biomarker.all [below](https://github.com/Code-the-Dream-School/health-keeper/pull/23/files#diff-f7fcfdbad10e32bb265e9ba7415ebc17190f6dbd815cca2caeed74a00af4b6efR114)?